src/app/core/services/managed-subscription.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Subscription, SubscriptionLike } from 'rxjs'
+
+export class ManagedSubscription implements SubscriptionLike {
```

Comment:
That it triggers a callback, when the user calls `unsubscribe` on it.

Reply:
Currently don't have any other suggestions. Was just curious.

But since you mentioned it, how does the Subscription become voided?
Also you have put the responsibility of unsubscribing on the "caller" that wants to listen so they are basically the "managers" of unsubscribing and they should be able to do that with just a regular subscription?

Reply:
When you call e.g. the RundownStateService for the first time and get a subscription, we create an internal BehaviourSubject. When a user calls unsubscribe on the given Subscription, we don't get a notification for it, such that we can clean up and remove the BehaviourSubject if it no longer has any subscriptions.

Reply:
But do we rellay need to kill the BehaviourSubject just because no one is listening?

We have to create it again if someone else wants to start listening.
Without having looked at the implementation of `RundownStateService`, i would assume that you go fetch the Rundown from the database when the BehaviourSubject is created.

So if I am the last Subscription and I unsubscribe, then we kill the BehaviourSubject, but then I want to subscribe 10 ms after I unsubscribe (page refresh), then you have to go call the backend for data you just discarded?

Personally, I think it's fine that the `RundownStateService` has one BehaviourSubject for each Rundown that just lives the lifespan of `RundownStateService`.